File: leetcode/3095/shortest_subarray_with_or_at_least_k.py
# 39ms | 16.66mb
# 97.71% | 22.44%
class Solution:
    def minimumSubarrayLength(self, nums: list[int], k: int) -> int:
        if max(nums) >= k:
            return 1

        smallest = -1

        for size in range(2, len(nums)+1):
            for subarray_i in range(len(nums) - size + 1):
                subarray = nums[subarray_i:subarray_i+size]
                or_result = 0
                for elem in subarray:
                    or_result = or_result | elem
                if or_result >= k:
                    return size

        return smallest

File: leetcode/3095/shortest_subarray_with_or_at_least_k.cpp
// 17ms | 28.11mb
// 5.36% | 5.20%
class Solution {

public: 
  int minimumSubarrayLength(vector<int>& nums, int k) {
    if (*max_element(nums.begin(), nums.end()) >= k) {
      return 1;
    }

    for (int size = 2; size <= nums.size(); size++) {
      for (int index = 0; index <= nums.size() - size; index++) {
        vector<int> subarray(nums.begin() + index, nums.begin() + index+size);
        int bitwise = 0;
        for (int elem : subarray) {
          bitwise |= elem;
        }
        if (bitwise >= k) {
          return size;
        }
      }
    } 

    return -1;
  }
};
